confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java

```diff
@@ -590,7 +591,54 @@ public int getUnknownDataNodesNum() {
     return allDataNodes.size();
   }
 
-  public void addMetrics() {
+  /**
+   * Get the LeaderCount of each DataNodeId
+   *
+   * @return Map<DataNodeId, LeaderCount>
+   */
+  public Map<Integer, Integer> getLeadershipCountByDatanode() {
+    Map<Integer, Integer> idToCountMap = new ConcurrentHashMap<>();
+
+    regionGroupCacheMap.forEach(
+        (consensusGroupId, regionGroupCache) -> {
+          Integer count = idToCountMap.get(regionGroupCache.getLeaderDataNodeId());
+          if (count == null) {
+            idToCountMap.put(regionGroupCache.getLeaderDataNodeId(), 1);
+          } else {
+            idToCountMap.put(regionGroupCache.getLeaderDataNodeId(), count + 1);
+          }
+        });
+    return idToCountMap;
+  }
+
+  public void addLeaderCount() {
+    Map<Integer, Integer> idToCountMap = getLeadershipCountByDatanode();
+    getNodeManager()
+        .getRegisteredDataNodes(-1)
+        .forEach(
+            dataNodeInfo -> {
+              TDataNodeLocation dataNodeLocation = dataNodeInfo.getLocation();
+              int dataNodeId = dataNodeLocation.getDataNodeId();
+              if (idToCountMap.containsKey(dataNodeId)) {
+                String name =
```

Comment:
I think this line can be extracted to a common method.
It has been used 5 times.

![image](https://user-images.githubusercontent.com/6756545/183580729-37d638d8-884d-44cb-816b-9ae353375566.png)

Reply:
Okay, I'll add it in the tool class.